File: php/chinese-demo/service/httpclient.php
<?php
/**
 * 发送http的json请求
 *
 * @param $url 请求url
 * @param $jsonStr 发送的json字符串
 * @return array
 */

function http_post_json($url, $jsonStr)
{
    print($jsonStr);
    
    $ch = curl_init();
    curl_setopt($ch, CURLOPT_POST, 1);
    curl_setopt($ch, CURLOPT_URL, $url);
    curl_setopt($ch, CURLOPT_POSTFIELDS, $jsonStr);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
    curl_setopt($ch, CURLOPT_HTTPHEADER, array(
            'Content-Type: application/json; charset=utf-8',
            'Content-Length: ' . strlen($jsonStr)
        )
    );

    $response = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);

    curl_close($ch);
 
    return array($httpCode, $response);
}

?>

File: python/english-demo/xpyunopensdk/service/httpclient.py
import requests
import json
import xpyunopensdk.model.model as model

"""
 send http request with json body
 @param $url 
 @param $jsonStr
 @return array
"""


def http_post_json(url, body):
    headers = {
        "Content-Type": "application/json; charset=UTF-8"
    }

    response = requests.post(url, data=json.dumps(body), headers=headers)

    result = model.XPYunResp
    result.httpStatusCode = response.status_code
    jsonObj = json.loads(response.text)
    result.content = model.XPYunRespContent
    result.content.code = jsonObj["code"]
    result.content.msg = jsonObj["msg"]
    # result.content.data = jsonObj["data"]
    result.content.serverExecutedTime = jsonObj["serverExecutedTime"]

    return result

